exploration/selection-declaration.md

```diff
@@ -356,3 +356,69 @@ and a data model error otherwise.
   Removes some self-documentation from the pattern.
 - Requires the pattern to change if the selectors are modified.
 - Limits number of referenceable selectors to 10 (in the current form)
+
+### Hybrid approach: Match may mutate, no duplicates
+
+In this alternative, in a `.match` statement:
+
+1. variables are mutated by their annotation
+2. no variable can be the operand in two selectors
+
+This keeps most messages more concise, producing the expected results in Example 1.
+
+#### Example 1
+
+```
+.match {$count :integer}
```

Comment:
I know it is already submitted, and we are balloting on it.
But this is risky.

Think of a parameter of type `Person`, with first / last name, DOB, gender, etc.

```
.match {$host :gender}
female {You got an invitation from {$host} to her party.}
...
```

One would really expect something like "... from Maria to her ..."
But because the `.match` "binds" the parameter with `:gender` now we get "... from female to her ..."
Can be solved with a local declaration.
But surprising.

So we fix the plural (`:number`) surprise (select on something and render something else), and we replace it with a different surprise.

Reply:
(as individual)

Presumably `:gender` as a selector is not a formatting function. The default formatting function for `$host` of type "person" would be implementation defined and presumably not be the `:gender` selector, despite the annotation. `:number` or `:integer` are interesting because they are both selectors and formatters. I can imagine other values behaving like that as well (function defaulting is currently implementation defined)

Compound objects, such as person names or measurements, probably are complicated because of the reasonable desire to do query-style selection (e.g. the plural category of a currency value's _number_ or, conversely, select based on the currency value's _currencyCode_).

We haven't fully baked "resolved value" yet and this is part of that discussion.

One last thought: is this really that different from some of the other options being balloted? This consideration would go in the pro/con buckets for each.
---

(chair hat ON)

Note that we are balloting selection's relationship to declarations, but the voting is only advisory to a technical conversation. Also, you are encouraged to vote against solutions you don't like and to discuss them on the discussion thread.